lib/logflare_web/live/query_live.ex

```diff
@@ -163,37 +161,41 @@ defmodule LogflareWeb.QueryLive do
   end
 
   def handle_params(params, _uri, socket) do
-    query_string = params["q"] || socket.assigns.query_string
+    socket =
+      if !socket.assigns.query_string,
+        do: assign(socket, :query_string, params["q"]),
```

Comment:
There may be a potential bug lurking here. If `params["q"]` is nil, then the `socket.assigns.query_string` will be `nil`. This means `!= ""` below will be true, and you will send a message to parse it. Parsing calls `Endpoints.parse_query_string(query_string)` which, at least according to its typespec, expects a string.

It may make sense to normalize the query string here to be either always an empty string or always nil. Perhaps the second is more idiomatic?

Reply:
👍  adjusted to normalize in the handle_params/2, which is much cleaner as the query string determination is now localised instead of spread out in the both mount/3 and handle_params/2